digdag-core/src/main/java/io/digdag/core/log/LocalFileLogServerFactory.java

```diff
@@ -193,10 +217,30 @@ public void close()
             {
                 try {
                     output.close();
+
+                    if (realtimeFollow) {
+                        // Delaying for deterrence duplicate output in digdag log --follow
+                        try {
+                            Thread.sleep(1000);
+                        }
+                        catch (InterruptedException ex) {
+                            Thread.currentThread().interrupt();
+                        }
```

Comment:
Thank you for comment!

The reasone is here.
https://github.com/treasure-data/digdag/pull/700#discussion_r176960045

(It may be confusing...)

Reply:
Got it. Thank you for pointing it. Do you already know why log messages are duplicated? If you don't know, we'd better to take a look the behavior before we start improving it.

Reply:
Thanks! Let me think about it..

Reply:
@alu can you share small workflow example to reproduce duplicated logs?

Reply:
@muga 

My example is [here](https://github.com/treasure-data/digdag/pull/700#issuecomment-353217730).

Reply:
@alu Thank you. Got it and understand the behavior. Can you tell me why you need to print logs from gzip file instead of plain text in your duplicate scenario?

Reply:
@muga Because `/api/logs/{id}/files` returns gzip handles as final, so it is will be printed.

1. `example.log` will be created after the workflow started
2. `example.log` contents will be printed on console as real time (the api returns `example.log` handle)
3. The workflow completed
4. `example.log` will be compressed into gzip as `example.gzip`
5. `example.log` will be removed
6. The api returns `example.gzip` handle
7. Digdag client thinks `example.gzip` handle is new log file, so print it (duplicated!)

Reply:
Thanks @alu. Understand that it's not trivial for `LocalFileLogServerFactory` to recognize that files are already read or not. I think that `TaskLogWatcher` or `TaskLogState` can check that. What do you think that the client filters and skips `.gzip` files to show logs if the client uses `follow` option?

Reply:
@muga Thanks!

Sounds good to
>  .gzip files to show logs if the client uses follow option?

And implemented it in [this commmit](https://github.com/alu/digdag/commit/683f70d9e773f5d8dad4abaf30088eafa79793a3).

Bu we got new issues...

- `log-server.realtime_follow` is optional, so the new logick should judgment it was enabled (how is do it?)
- If an user exec `logs -f` when after finished a task, the user can't show finished task logs. For example
   1. task1 finished
   2. task1 logs compressed
   2. task2 started
   3. an user exec `logs -f` -> can't show task1 logs because the logs already compressed

Reply:
@muga Can i get some advices for my issues?

Reply:
> log-server.realtime_follow is optional, so the new logick should judgment it was enabled (how is do it?)

Not 100% sure but, my idea was that a watcher can filter and treat both of .log.gz and .log files as same file in `TaskLogState` to avoid duplication by new .log.gz file. If this idea would work for you, your second issue could be solved I think. What do you think?

Reply:
@muga 

I improve my code solved my 2 issues!
Skip show gzip log when a task log already shown as txt format!

How about it!?

Thanks!